docs/md/sink.md

```diff
@@ -0,0 +1,132 @@
+will be created as. The entity type must exist prior to workload creation and
+must be a table in the workspace.
+
+#### `identifier`
+The `identifier` is the name of a pipeline output that should be used as the 
+name of each newly created entity. 
+
+Example - Let's say the pipeline you're running has an output called
+"sample_name" that uniquely identifies the inputs and outputs to that pipeline.
+By setting `"identifier": "sample_name"` in the sink configuration, entities
+will be created using the "sample_name" as the entity name.
+
+Note that when two sets of pipeline outputs share the same "identifier" value,
+the first set of outputs will be overwritten by the second in the workspace.
+
+#### `fromOutputs`
+`fromOutputs` configures how to create new entities from pipeline outputs by 
+mapping the output names to attributes in the `entityType`. Note that all 
+attribute names must exist in the entityType before the workload  creation.
+
+`fromOutputs` allows a small about of flexibility in how to construct an entity
+and supports the following relations:
+
+- `"attribute": "output"`
+  Direct mapping from an output to an attribute
+
+- `"attribute": ["output0", "output2"]`
+  Make an attribute from a list of pipeline outputs.
+
+- `"attribute": "$value"`
+  Use the literal "value" for an attribute.
+
+## Developer Guide
+A sink is one satisfying the `Sink` protocol as below:
+```clojure
+(defprotocol Sink
+  (update-sink!
+    ^Sink
+    [^Queue upstream ;; The queue to sink outputs from
+     ^Sink  sink     ;; This sink instance
+    ]
+    "Update the internal state of the `sink`, consuming objects from the
+     upstream `Queue`, performing any external effects as required.
+     Notes:
+     - The `Sink` and `Queue` are parameterised types and the `Queue`'s 
+       parameterisation must be convertible to the `Sink`s.
+     - Where possible, `Sink` implementations should minimise blocking and use
+       an internal queue to schedule and manage long-running computations."))
+```
+
+!!! info
+    The `Sink` protocol is currently implemented by the `update-sink!`
+    multimethod.
+
+To be used in a workload, a `Sink` implementation should satisfy the processing
+`stage` protocol and the `to-edn` multimethod in addition to the following
+multimethods specific to sinks:
+```clojure
+(defmulti create-sink
+  "Create a `Sink` instance using the database `transaction` and configuration
+   in the sink `request` and return a `[type items]` pair to be written to a
+   workload record as `sink_type` and `sink_items`.
+   Notes:
+   - This is a factory method registered for workload creation.
+   - The `Sink` type string must match a value of the `sink` enum in the
+     database schema.   
+   - This multimethod is type-dispatched on the `:name` association in the
+     `request`."
+  (fn ^[^String ^String] 
+      [^Connection         transaction  ;; JDBC Connection
+       ^long               workload-id  ;; ID of the workload being created
```

Comment:
Should the workload-id be hinted as a UUID instead?

Reply:
Nope - it's a long.